Tasks/AzureResourceManagerTemplateDeploymentV3/operations/DeploymentScopeBase.ts

```diff
@@ -62,6 +62,11 @@ export class DeploymentScopeBase {
         if (this.deploymentParameters.properties["mode"] === "Validation") {
             return this.validateDeployment();
         } else {
+            try {
+                await this.validateDeployment();
+            } catch (error) {
+                tl.warning(tl.loc("TemplateValidationFailure", error));
```

Comment:
I think we should log the warning only in case the deployment fails. Otherwise people might not like the warning.

Reply:
or we can do this later only if people complain

Reply:
There are couple of errors being printed from within validateDeployment function. So I think we can keep it like this for now